spec-dtslint/operators/bufferToggle-spec.ts

```diff
@@ -23,7 +23,8 @@ it('should enforce types', () => {
 });
 
 it('should enforce type of openings', () => {
-  const o = of(1, 2, 3).pipe(bufferToggle('a', () => of('a', 'b', 'c'))); // $ExpectError
+  const o = of(1, 2, 3).pipe(bufferToggle('a', () => of('a', 'b', 'c'))); // $ExpectType Observable<number[]>
+  const o2 = of(1, 2, 3).pipe(bufferToggle('a', (x: number) => of('a', 'b', 'c'))); // $ExpectError
```

Comment:
Is this a regression or was it always supposed to word but limited by TS?

Reply:
Before `bufferToggle` was only taking `SubscribableOrPromise`, which isn't accurate. It should have been `ObservableInput`. Once that was in, `string`, which is an `Iterable`, was valid, making the above test inaccurate.

Then I added the bit to make sure the type passed to the closing selector was accurate.